File: src/main/java/pl/yalgrin/playnite/simplesync/dto/ChangeDTO.java
package pl.yalgrin.playnite.simplesync.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import pl.yalgrin.playnite.simplesync.enums.ObjectType;
import pl.yalgrin.playnite.simplesync.utils.ToStringUtils;

@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
public class ChangeDTO extends AbstractDTO {
    @JsonProperty("Id")
    private Long id;
    @JsonProperty("Type")
    private ObjectType type;
    @JsonProperty("ClientId")
    private String clientId;
    @JsonProperty("ObjectId")
    private Long objectId;
    @JsonProperty("ForceFetch")
    private boolean forceFetch;

    @Override
    public String toString() {
        return ToStringUtils.createBuilder(this)
                .append("id", id)
                .append("type", type)
                .append("clientId", clientId)
                .append("objectId", objectId)
                .append("forceFetch", forceFetch)
                .toString();
    }
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/change/dto/ChangeDTO.kt
package pl.yalgrin.playnite.simplesync.change.dto

import com.fasterxml.jackson.annotation.JsonProperty
import pl.yalgrin.playnite.simplesync.common.dto.AbstractDTO
import pl.yalgrin.playnite.simplesync.common.enums.ObjectType
import pl.yalgrin.playnite.simplesync.common.utils.ToStringUtils
import java.io.Serial

data class ChangeDTO(
    var id: Long? = null,
    var type: ObjectType? = null,
    var clientId: String? = null,
    var objectId: Long? = null,
    @param:JsonProperty("isForceFetch")
    @get:JsonProperty("isForceFetch")
    @field:JsonProperty("isForceFetch")
    var isForceFetch: Boolean = false
) : AbstractDTO() {

    override fun toString(): String {
        return ToStringUtils.createBuilder(this)
            .append("id", id)
            .append("type", type)
            .append("clientId", clientId)
            .append("objectId", objectId)
            .append("isForceFetch", isForceFetch)
            .toString()
    }

    companion object {
        @Serial
        private const val serialVersionUID = 6362405151054072827L
    }
}